src/domain/usecases/ClosePatientsUseCase.ts

```diff
@@ -25,16 +25,43 @@ export class ClosePatientsUseCase {
 
         const payload = await this.programsRepository
             .get({ programId, orgUnitsIds, startDate, endDate })
-            .then(trackedEntities =>
-                this.filterEntities(
-                    trackedEntities,
+            .then(async trackedEntities => ({
+                teis: await this.programsRepository.getTeis(trackedEntities.map(tei => tei.trackedEntity)), //TEMPORAL FIX
+                trackedEntities,
+            }))
+            .then(({ trackedEntities, teis }) => {
+                const fixedTrackedEntities = trackedEntities.map(entity => {
+                    const teiWithRealOrgUnits = teis.find(
+                        tei => tei.trackedEntityInstance === entity.trackedEntity
+                    );
+                    const newEnrollments = entity.enrollments.map(enrollment => {
+                        const realEnrollment = teiWithRealOrgUnits?.enrollments.find(
+                            realEnrollment => realEnrollment.enrollment === enrollment.enrollment
+                        );
+
+                        return {
+                            ...enrollment,
+                            orgUnit: realEnrollment ? realEnrollment.orgUnit : enrollment.orgUnit,
+                            orgUnitName: realEnrollment
+                                ? (realEnrollment as any).orgUnitName
```

Comment:
`any`. 

1) I see that's because field `orgUnitName` is not defined in d2-api. Tipically, we'd add the field to the type `TrackedEntityInstancePick` and, when necessary, case it: `tei as unknown as TrackedEntityInstancePick`. This way we cast not usage but on definition.

2) Not sure we need orgUnitName, I guess it can be removed from the payload and it will use orgUnit ID.